pep/002-mixin-versioning.md

```diff
@@ -0,0 +1,352 @@
+---
+title: "Mixin Versioning"
+number: "NNN"
+status: "provisional"
+owner: "@carolynvs"
+authors:
+  - "@carolynvs"
+  - "@sestegra"
+created: "2021-02-21"
+type: "feature"
+---
+
+## Abstract
+
+Bundles should be able to specify not only the mixin used, but its version and source.
+Porter should use that information to automatically download the mixins declared by the bundle.
+
+
+## Motivation
+
+Currently bundle authors manage the mixins used within a bundle by hand. 
+They must manually install each mixin using homegrown scripts. 
+Unfortunately the mixin cache is stored in PORTER_HOME, it is impossible to work on bundles that use different versions of the same mixin.
+Porter really needs to manage the mixin cache and appropriately communicating with the correct mixin version targeted by the bundle.
+
+* Mixin authors are impacted by this change because they will need to adjust their mixin declaration schema to allow for declaring the mixin version.
+* Bundle authors are impacted by the new syntax for declaring mixins and no longer managing the bundle cache by hand.
+* End users will now be able to see the exact version of the mixins used by a bundle.
```

Comment:
The first time they use the new version, additional bandwidth/time to download new versions because meta information will be updated?

Reply:
This might be the impact to the bundle author, I'm not sure exactly.

Reply:
I think a more clear explanation is that mixin authors will need to release new versions of their mixin supporting this change. Let me add that.